File: my-folder/0078-subsets/solution.java
class Solution {

    List<List<Integer>> res = new ArrayList<>();

    public List<List<Integer>> subsets(int[] nums) {
        backtrack(nums, 0, new ArrayList<>());
        return res;
    }

    private void backtrack(int[] nums, int index, List<Integer> temp) {
        res.add(new ArrayList<>(temp));
        for (int i = index; i < nums.length; i++) {
            temp.add(nums[i]);
            backtrack(nums, i + 1, temp);
            temp.remove(temp.size() - 1);
        }
    }
}

File: my-folder/0078-subsets/solution.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.res = []
        def backtrack(nums, index, temp):
            self.res.append(temp[:])
            for i in range(index, len(nums)):
                temp.append(nums[i])
                backtrack(nums, i + 1, temp)
                temp.pop()
        backtrack(nums, 0, [])
        return self.res
